lib/processAllArtifacts.js

```diff
@@ -12,14 +13,59 @@ var readFile = util.promisify(fs.readFile)
 var writeFile = util.promisify(fs.writeFile)
 var readdir = util.promisify(fs.readdir)
 
-module.exports = async function (inputArtifactsPath, outputArtifactsPath, networksPath) {
-  let inputDirError = false
+const updateArtifactFiles = (inputArtifactsPath, outputArtifactsPath, networkConfigs) => {
+  return new Promise(async (resolve, reject) => {
```

Comment:
instead of creating a new promise, you could just declare `updateArtifactFiles` as an async function, then `await` on it below where you use it.

It might be cleaner- because I'm not sure how errors would be handled here.